crates/oci-tar-builder/src/lib.rs

```diff
@@ -10,14 +10,76 @@ use oci_spec::image::{
     DescriptorBuilder, ImageConfiguration, ImageIndexBuilder, ImageManifestBuilder, MediaType,
     PlatformBuilder, SCHEMA_VERSION,
 };
+use oci_wasm::{WasmConfig, WASM_ARCHITECTURE};
 use serde::Serialize;
 use sha256::{digest, try_digest};
-#[derive(Debug, Default)]
-pub struct Builder {
-    configs: Vec<(ImageConfiguration, String)>,
+#[derive(Debug)]
+pub struct Builder<C: OciConfig> {
+    configs: Vec<(C, String, MediaType)>,
     layers: Vec<(PathBuf, String)>,
 }
 
+pub trait OciConfig {
+    fn os(&self) -> String;
+    fn architecture(&self) -> String;
+    fn layers(&self) -> Vec<String>;
+    fn to_string(&self) -> String;
+}
+
+impl OciConfig for ImageConfiguration {
+    fn os(&self) -> String {
+        self.os().to_string()
+    }
+
+    fn architecture(&self) -> String {
+        self.architecture().to_string()
+    }
+
+    fn layers(&self) -> Vec<String> {
+        self.rootfs().diff_ids().to_vec()
+    }
+
+    fn to_string(&self) -> String {
+        self.to_string_pretty().unwrap()
+    }
+}
+
+impl OciConfig for WasmConfig {
+    fn os(&self) -> String {
+        self.os.to_string()
+    }
+
+    fn architecture(&self) -> String {
+        WASM_ARCHITECTURE.to_string()
+    }
+
+    fn layers(&self) -> Vec<String> {
+        self.layer_digests.clone()
+    }
+
+    fn to_string(&self) -> String {
+        serde_json::to_string_pretty(self).unwrap()
+    }
+}
+
+impl Default for Builder<WasmConfig> {
+    fn default() -> Self {
+        Self {
+            configs: Vec::new(),
+            layers: Vec::new(),
+        }
+    }
+}
+
+impl Default for Builder<ImageConfiguration> {
+    fn default() -> Self {
+        Self {
+            configs: Vec::new(),
+            layers: Vec::new(),
+        }
+    }
+}
```

Comment:
nit: remove as the above suggestion derives `Default` trait

Reply:
I tried this suggestion but default marcro doesn't work since we have two different types, one of which doesn't have a default implementation